### DISCUSSION TITLE
Restructure repo

### DISCUSSION THREAD

**Discussion by haampie** [spack/spack#23811] [Ideas]
### Rationale

With over 5k packages navigating the builtin repo is quite cumbersome, specifically through the GitHub UI:

>  Sorry, we had to truncate this directory to 1,000 files. 4,582 entries were omitted from the list. 

### Description

Some package managers choose to create subdirs for the first letter:

``a/ b/ c/ ... z/``

Or even

```aa/ ab/ ac/ ... ba/ bb/ ... zz/```

Or

```a/a/ a/b/ ... z/z/```

That would make browsing through github a bit easier.

**Comment by adamjstewart** [comment]
Another idea I've had would be to put language-specific packages in subdirectories, like `py/`, `r/`, etc. However, we now have ~1K R packages, so this would give us the same issue. Of course, we would also run into this issue with `r/` and `p/`.

Current workaround is to manually type the package name in the browser URL, but I understand that this is cumbersome. Most other package managers and repos like PyPI seems to use the structure you propose.

**Comment by adamjstewart** [comment]
P.S. I wonder if this should be a Discussion instead of an Issue, although no one seems to pay attention to Discussions...

**Comment by vsoch** [comment]
We do this in autamus! https://github.com/autamus/registry/tree/main/spack
